Rename packages from "ponyc-release" to "ponyc"

killerswan opened the pull request:
Simplifies a bunch of finicky little things.

Note, this would also require renaming (if possible) or creating new packages on Bintray with the name "ponyc" instead of "ponyc-release".  😅 

As noted on #1713.

SeanTAllen commented:
@killerswan this requires changes to the README for installation yes?

also, are you handling the bintray change?

if we rename, that will break existing setups that folks might have right? for apt source etc?

killerswan commented:
It has the README changes, I think.

I can do the Bintray changes.   To reduce breaking any people's scripts which require a specific old release of "ponyc-release" we can do two things:
* leave the "ponyc-release" Bintray-package as is (since I can't see how to rename, only add/delete), along side the new "ponyc" Bintray-package, and
* make the new "ponyc" RPM and DEB packages provide "ponyc-release" (I'll do this in a moment).

The Fedora .REPO files, for example, do not mention individual packages.  For example, the one Bintray suggests you create ([here, clicking the button](https://bintray.com/pony-language/ponyc-rpm)), does not mention "ponyc-release", "ponydep-ncurses-libs-5", or "ponydep-ncurses-libs-5-compat", but provides all three packages:
```
#bintraybintray-pony-language-ponyc-rpm - packages by pony-language from Bintray
[bintraybintray-pony-language-ponyc-rpm]
name=bintray-pony-language-ponyc-rpm
baseurl=https://dl.bintray.com/pony-language/ponyc-rpm
gpgcheck=0
repo_gpgcheck=0
enabled=1
```

Still, I am hesitant to merge this late on a Friday. 😅

killerswan commented:
As I said I'd do:
* I've added another commit (and rebased a bit) to provide package conflict/provides for RPM/DEB.
* I've now created "ponyc" packages in each of the four repos [here](https://bintray.com/pony-language).

SeanTAllen commented:
@killerswan we have enough "fix compilation on XYZ" platform things in the changelog to justify another release especially as the mtune to improve rpm and deb performance is part of that. So, if we were needing to do a release as part of this, I think its reasonable to do soon. 

On all things release, I trust you and what you are doing. 

- Should I merge this? (assuming yes as there is no DO NOT MERGE tag)
- When its merged, do we want to do a release?
- Do we need to inform people of a change to the rpm/apt source locations?

killerswan commented:
😸 

In order, then:

1. Yeah.  I'm now awake and will be around the computer until later tonight, so we can fix things if they're not working yet, so merge away!
1. After we look at the logs from it's master build, yeah, we can do another release whenever you want.
1. Nope.  This shouldn't change the location of any package repository.  Just the package name itself.

Currently, before and after this pull req., there are four package repositories, which shouldn't change here, under https://bintray.com/pony-language/ : 
* ponyc-debian,
* ponyc-rpm,
* ponyc-source, and
* ponyc-win.
